global_logger.py

```diff
@@ -1,23 +1,174 @@
-import logging
 import logging.handlers
 import os
+import time
+from enum import auto, Enum
+from functools import wraps
+from typing import Optional
 
+# Set up logging level constants
+CRITICAL = logging.CRITICAL  # 50
+ERROR = logging.ERROR  # 40
+WARNING = logging.WARNING  # 30
+INFO = logging.INFO  # 20
 VERBOSE = 15
-verbose_level = 'VERBOSE'
-log_file = './logs/logon_server.log'
+DEBUG = logging.DEBUG  # 10
 
-log_formatter = logging.Formatter("%(levelname)-7.7s %(process)-12.12d %(processName)-12.12s %(threadName)-12.12s %(asctime)s: %(message)s")
+# Setup internal vars
+_verbose_level_name = 'VERBOSE'
+_log_file = './logs/logon_server.log'
+
+# noinspection SpellCheckingInspection
+_log_formatter = logging.Formatter(
+    "%(levelname)-7.7s %(process)-12.12d %(processName)-12.12s %(threadName)-12.12s %(asctime)s: %(message)s")
 logger = logging.getLogger()
-logging.addLevelName(VERBOSE, verbose_level)
+logging.addLevelName(VERBOSE, _verbose_level_name)
+
+# Make directory (if it doesn't exist)
+os.makedirs(os.path.dirname(_log_file), exist_ok=True)
 
-# file_handler = logging.FileHandler("{0}/{1}.log".format(logPath, fileName))
-file_handler = logging.handlers.WatchedFileHandler(os.environ.get('LOGFILE', log_file))
-file_handler.setFormatter(log_formatter)
+# Setup handlers
+file_handler = logging.handlers.WatchedFileHandler(filename=os.environ.get('LOGFILE', _log_file), mode='a+')
+file_handler.setFormatter(_log_formatter)
 console_handler = logging.StreamHandler()
-console_handler.setFormatter(log_formatter)
+console_handler.setFormatter(_log_formatter)
 logger.addHandler(file_handler)
 logger.addHandler(console_handler)
-
-#logger.setLevel(logging.INFO)
-#logger.setLevel(verbose_level)
 logger.setLevel(logging.DEBUG)
+
+
+class _WrapperType(Enum):
```

Comment:
This is for internal use only, hence the leading underscore. 

I am using this to reduce repeated code in `logged_method`, `logged_class_method`, and `logged_fucntion`.